Packs/LogrhythmV2/Integrations/LogRhythmV2/LogRhythmV2.py

```diff
@@ -0,0 +1,2492 @@
+
+        if recipient:
+            query.append(self.generate_query_item(32, 4, str(recipient)))
+
+        if hash_:
+            query.append(self.generate_query_item(138, 4, str(hash_)))
+
+        if url:
+            query.append(self.generate_query_item(42, 4, str(url)))
+
+        if process_name:
+            query.append(self.generate_query_item(41, 4, str(process_name)))
+
+        if object_:
+            query.append(self.generate_query_item(34, 4, str(object_)))
+
+        if ipaddress:
+            query.append(self.generate_query_item(17, 5, ipaddress))
+
+        # Search and get TaskID
+        data = {
+            "maxMsgsToQuery": int(max_message),
+            "logCacheSize": 10000,
+            "queryTimeout": int(query_timeout),
+            "queryRawLog": True,
+            "queryEventManager": False,
+            "dateCriteria": {
+                "useInsertedDate": False,
+                "lastIntervalValue": int(number_of_days),
+                "lastIntervalUnit": 4
+            },
+            "queryLogSources": [],
+            "queryFilter": {
+                "msgFilterType": 2,
+                "isSavedFilter": False,
+                "filterGroup": {
+                    "filterItemType": 1,
+                    "fieldOperator": 1,
+                    "filterMode": 1,
+                    "filterGroupOperator": 0,
+                    "filterItems": query
+                }
+            }
+        }
+
+        response = self._http_request('POST', 'lr-search-api/actions/search-task', json_data=data, headers=headers)
+
+        return response
+
+    def generate_query_item(self, filter_type, value_type, value):
+        if value_type == 2:
+            value = int(value)
+        elif value_type == 5:
+            value = str(value)
+        else:
+            value = {
+                "value": value,
+                "matchType": 2
+            }
+
+        query = {
+            "filterItemType": 0,
+            "fieldOperator": 1,
+            "filterMode": 1,
+            "values": [
+                {
+                    "filterType": filter_type,
+                    "valueType": value_type,
+                    "value": value
+                }
+            ]
+        }
+
+        return query
+
+    def get_query_result_request(self, task_id, page_size):
+        headers = self._headers
+        data = {
+            'data': {
+                'searchGuid': task_id,
+                'search': {
+                    'sort': [],
+                    'groupBy': [
+                        'string'
+                    ],
+                    'fields': []
+                },
+                'paginator': {
+                    'origin': 0,
+                    'page_size': int(page_size)
+                }
+            }
+        }
+
+        response = self._http_request('POST', 'lr-search-api/actions/search-result', json_data=data, headers=headers)
```

Comment:
if not given, the function will the take the property anyway.
```suggestion
        response = self._http_request('POST', 'lr-search-api/actions/search-result', json_data=data)
```

Reply:
done